src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,110 @@
+package duke;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Represents a storage which allows for reading and writing of tasks to a file.
+ */
+public class Storage {
+    private static final Pattern STORE_FORMAT = Pattern.compile("\\[(?<type>\\S)] \\[(?<done> |X)] (?<arguments>.*)");
+    private final String fileName;
```

Comment:
should a constant variable be in cammelcase?

Reply:
Constant names must be all uppercase using underscore to separate words, as mentioned in the java coding standard.